Sources/MongoSwift/BSON/Document.swift

```diff
@@ -157,17 +164,34 @@ extension Document {
 
                 // otherwise, we just create a new document and replace this key
             } else {
-                // TODO: SWIFT-224: use va_list variant of bson_copy_to_excluding to improve performance
-                var newSelf = Document()
+                guard let iter = DocumentIterator(forDocument: self) else {
+                    throw Document.BSONBufferTooSmallError
+                }
+
+                var keysBefore: [String] = []
+                var keysAfter: [String] = []
```

Comment:
based on these names, I would've guessed that `keysBefore` referred to the "keys that come before the set one", but based on their usage it seems to be the opposite. To reduce the confusion, could you just add comments to each of these indicating what they refer to?

Reply:
Done.